### PR TITLE
Fix delegating ctor example

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

**Comment by bernhardmgruber** [comment]
How does this fix the example? I think both versions (the one before and your suggested one) are valid use cases. I would rather prefer to either *add* your version, or add a comment that both situations are possible.

Delegating constructors are about avoiding code duplications. That should be the main point. How ctors delegate to each other is less important.

**Comment by eguiraud** [comment]
Unfortunately the original example is not a valid use of delegating constructors (although I wish it was, because it makes absolute sense): https://godbolt.org/z/MTsz4n3x6

File: talk/objectorientation/constructors.tex [hunk 1]
```diff
@@ -204,8 +204,8 @@
     \begin{cppcode}
       struct Delegate {
         int m_i;
-        Delegate() { ... complex initialization ...}
-        Delegate(int i) : Delegate(), m_i(i) {}
+        Delegate(int i) : m_i(i) {}
```
Comment (sponce) [inline talk/objectorientation/constructors.tex:207]:
```suggestion
        Delegate(int i) : m_i(i) {
          ... complex initialization ...
        }
```

**Review by sponce — APPROVED** [review]
Damn, I'm learning something... As every day with C++ ;-)

Anyway, I agree on the change but would still leave the "complex initialization" part just to insist on code reuse, which is the coere feature as @bernhardmgruber was pointing it

**Comment by bernhardmgruber** [comment]
Wow, I did not know that! Cool!

**Pull request merged**